addons/mass_mailing/views/mailing_mailing_views.xml

```diff
@@ -189,18 +189,20 @@
                         </group>
                         <notebook>
                             <page string="Mail Body" name="mail_body">
-                                <div class="mt-n2">
-                                    <field name="body_html" class="o_mail_body oe_read_only" widget="html"
+                                <field name="is_body_empty" invisible="1"/>
+                                <div class="mt-n2 oe_read_only" attrs="{'invisible': [('is_body_empty', '=', True)]}">
```

Comment:
So this will hide the "body_html" field, if the "body_html" field is empty.
Seems a little strange.

Even though picking a template will almost always initialize the body_html with some value, it doesn't feel like a good idea, unless I'm missing something.

Reply:
Yes, the goal is to avoid showing a blank `iframe` that takes up some screen space. If I do not hide it, there will be a empty space over the action helper (if we use `position: static`).

If the user picks a template that contains nothing, the preview will be hidden and the action helper will be shown if the user is in 'read only' mode.

I agree that having an empty 'body_html' does not imply that the user didn't picked a template (-> The user could pick a template that contains nothing). If we don't want to display the action helper when the user picked an empty template, we would have to add a new field to the model to indicate that a template has been selected.

**Do you see something that could go wrong if I hide the preview iframe when it's empty ?**

As it's confusing to mix the `visibility` attribute with the `oe_read_only` and `oe_edit_only` classes, here is a summary of the behavior of the view with these new changes:

When the user is in 'read mode':
- We hide the preview if the preview is empty.
- We hide the editor.
- We hide the action helper if the preview is not empty or if the emails have been marked as 'sending' or 'done'.

When the user is in 'edit mode':
- We hide the preview.
- We show the editor.
- We hide the action helper.